### PR TITLE
Return only a single license file from the analysis.

### PR DESCRIPTION
**Pull request by isoos**
https://github.com/dart-lang/pana/issues/92

### CODE REVIEW

File: lib/src/license.dart [hunk 1]
```diff
@@ -13,14 +13,14 @@ import 'package:path/path.dart' as p;
 import 'download_utils.dart';
 import 'model.dart';
 
-Future<List<LicenseFile>> detectLicensesInDir(String baseDir) async {
+Future<LicenseFile> detectLicensesInDir(String baseDir) async {
   final rootFiles = await Directory(baseDir).list().toList();
   final licenseCandidates = rootFiles
       .whereType<File>()
       .where(_isLicenseFile)
       .take(5) // only the first 5 files are considered
```
Comment (sigurdm) [inline lib/src/license.dart:21]: Could we change this to 1?

**Comment by isoos** [comment]
> My only concern is that we might want to prioritize the license files

Good point, I'll add that (similar to the ones we have for readme, changelog and example).

File: lib/src/license.dart [hunk 2]
```diff
@@ -11,55 +11,36 @@ import 'dart:io';
...
-  final licenses = await Future.wait(licenseCandidates.map(
-    (File file) {
-      final relativePath = p.relative(file.path, from: baseDir);
-      return detectLicenseInFile(file, relativePath: relativePath);
-    },
-  ));
-
-  licenses.sort((l1, l2) => Comparable.compare(l1.path, l2.path));
-  return licenses;
+Future<LicenseFile> detectLicensesInDir(String baseDir) async {
```
Comment (jonasfj) [inline lib/src/license.dart:17]: Rename the method if it's not plural anymore

File: lib/src/license.dart [hunk 3]
```diff
@@ -11,55 +11,36 @@ import 'dart:io';
...
+  for (final candidate in licenseFileNames) {
```
Comment (jonasfj) [inline lib/src/license.dart:18]: Historically, the `pub` tool has a allow license files with many different names... `pana` and `pub.dev` needs to continue to support such old packages.

But I'm fine with saying that:
 * if there is a `LICENSE` file, then it is always the license we show, and,
 * if there is no `LICENSE` file, then we reject the package when published.

Note, that for existing packages, the old license extraction logic needs to be preserved, does it not?

Comment (isoos) [reply lib/src/license.dart:18]: > the old license extraction logic needs to be preserved

With the new rules we'll lose 4 licenses detected out of 10k+ (latest stable versions):
- `LICENSE.html` (2 instances)
- `LICENSE.GPLv3`
- `LICENSE.MIT`

Supporting and hosting a free-form `.html` is not something we want to do or encourage.

The remaining `LICENSE.<name>` is debatable - do we want to keep supporting them?

Comment (isoos) [reply lib/src/license.dart:18]: Btw. the current logic now follows the same selection precedence as we have in https://github.com/dart-lang/pub-dev/pull/3908

**Pull request merged**